### DISCUSSION TITLE
save two bits of data for each option in the datasource

### DISCUSSION THREAD

**Discussion by james-whittington1** [leekelleher/umbraco-contentment#413] [Question] [answered]
I am trying to create an experience that shows options with an associated score

eg favoutite sports:
 Football: 5
 Basketball: 10
 Hockey: 4
 
 
 I get the sport names from an injectable service and have made a custom data source which worked perfectly to present them as a list, but I am having trouble figuring out how to add a value to each of them. I want to use the custom angular template to give a range slider but don't see a way to save the value + the name of the sport. 
 
 Is the repeatable text box input the only one that will work for this?

**Comment by leekelleher** [comment, marked as answer]
@james-whittington1 Interestingly, a similar question came up on [the Discord chat last week](https://discord-chats.umbraco.com/t/22921346/editor-for-list-of-countries-with-a-value#d52c7f18-2ead-4b98-ab9e-711c0c454e2e), ([🛫](https://discord.com/channels/869656431308189746/1273283844761714698/1273608051827675170)).

In a nutshell, there's nothing available out of the box in Contentment to do this.

Options are, use Block List or Contentment's Content Blocks, then you have control over the element-type.
or to roll your own custom property-editor, then you'd have complete control.

My reply on Discord suggests that it could be possible using the internal Data Table editor to configure each row with Data List and Slider property-editors.

**Reply by james-whittington1** [reply]
Thanks, that is a great starting point.
